geomstats/geometry/positive_reals.py

```diff
@@ -0,0 +1,315 @@
+from geomstats.geometry.riemannian_metric import RiemannianMetric
+
+
+class PositiveReals(OpenSet):
+    """Class for the manifold of positive reals.
+
+    The real positive axis endowed with the Information geometry metric.
+
+    Parameters
+    ----------
+    scale : float
+        Scale of the positive reals metric.
+        Optional, default: 1.
+    """
+
+    def __init__(self, scale=1.0, **kwargs):
+        super().__init__(
+            dim=1,
+            embedding_space=Euclidean(1),
+            metric=PositiveRealsMetric(scale=scale),
+            **kwargs
+        )
+        self.scale = scale
+
+    @staticmethod
+    def belongs(point, atol=gs.atol):
+        """Check if a point is a positive real.
+
+        Evaluate if a point belongs to the positive real axis,
+        i.e. evaluate if its norm is lower than one.
+
+        Parameters
+        ----------
+        point : array-like, shape=[..., 1]
+            Points to be checked.
+        atol : float
+            Tolerance.
+            Optional, default: backend atol.
+
+        Returns
+        -------
+        belongs : array-like, shape=[...,]
+            Boolean denoting if point is a positive real.
+        """
+        point_shape = point.shape
+        is_scalar = len(point_shape) == 1
+        is_scalar = is_scalar or (len(point_shape) == 2 and point_shape[1] == 1)
+        is_real = gs.imag(point) == 0
+        point = gs.real(point)
+        is_positive = gs.array(
+            [[gs.all(point[i_point] > 0)] for i_point in range(point_shape[0])]
+        )
+        belongs = gs.logical_and(is_scalar and is_real, is_positive)
+        belongs = gs.reshape(belongs, (-1,))
+        return belongs
+
+    @staticmethod
+    def projection(point, atol=gs.atol):
+        """Project a point on the positive reals.
+
+        Parameters
+        ----------
+        point : array-like, shape=[..., 1]
+            Point in ambient manifold.
+        atol : float
+            Tolerance.
+            Optional, default: backend atol.
+
+        Returns
+        -------
+        projected : array-like, shape=[..., 1]
+            Projected point.
+        """
+        return gs.where(point <= 0, atol, point)
+
+    @staticmethod
+    def random_point(n_samples=1, bound=1.0, atol=gs.atol):
+        """Sample in the positive reals.
+
+        Parameters
+        ----------
+        n_samples : int
+            Number of samples.
+            Optional, default: 1.
+        bound : float
+            Bound of the interval in which to sample in the tangent space.
+            Optional, default: 1.
+        atol : float
+            Tolerance.
+            Optional, default: backend atol.
+
+        Returns
+        -------
+        samples : array-like, shape=[..., 1]
+            Points sampled in the positive reals.
+        """
+        size = (n_samples, 1) if n_samples != 1 else (1,)
+        samples = bound * gs.random.rand(*size)
+        samples = gs.maximum(samples, atol)
```

Comment:
why are we taking the maximum?

Reply:
We use the maximum because the point 0 is the boundary of the `PositiveReals` manifold, it is at an infinite distance from the other points. So we do not want the simulated random point to be to close to zero.
I have changed the `random_point` method: the simulated random point now has a uniform distribution between `atol` and `bound`.
Done.